app/src/main/java/com/khinthirisoe/bakingapp/ui/baking/view/BakingAdapter.kt

```diff
@@ -0,0 +1,76 @@
+package com.khinthirisoe.bakingapp.ui.baking.view
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.khinthirisoe.bakingapp.R
+import com.khinthirisoe.bakingapp.data.model.Recipe
+import com.khinthirisoe.bakingapp.ui.base.BaseViewHolder
+import com.khinthirisoe.bakingapp.ui.base.inflate
+import kotlinx.android.synthetic.main.list_baking.view.*
+
+class BakingAdapter(
+    private val context: Context,
+    private var recipes: MutableList<Recipe>?,
+    private val clickListener: BakingRecyclerViewClickListener
+) : RecyclerView.Adapter<BakingAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        return ViewHolder(parent.inflate(R.layout.list_baking))
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.onBind(position)
+    }
+
+    override fun getItemCount(): Int {
+
+        return if (recipes != null && recipes!!.size > 0) {
+            recipes!!.size
+        } else {
+            0
+        }
+    }
+
+    fun setRecipes(recipes: ArrayList<Recipe>) {
+        this.recipes = recipes
+        notifyDataSetChanged()
+    }
+
+    inner class ViewHolder(itemView: View) : BaseViewHolder(itemView) {
+
```

Comment:
@KhinThiriSoe write ViewHolder in a separate file. When your adapter and ViewHolder grow bigger, it will be really hard to manage if you put them in a single file